src/core/server/opensearch/opensearch_config.ts

```diff
@@ -131,6 +131,7 @@ export const configSchema = schema.object({
   healthCheck: schema.object({ delay: schema.duration({ defaultValue: 2500 }) }),
   ignoreVersionMismatch: schema.boolean({ defaultValue: false }),
   disablePrototypePoisoningProtection: schema.maybe(schema.boolean({ defaultValue: false })),
+  compression: schema.maybe(schema.boolean({ defaultValue: false })),
```

Comment:
```suggestion
  compression: schema.boolean({ defaultValue: false }),
```
When we have a default value, `maybe` is not needed. However, I like the idea @SuZhou-Joe proposed. I would prefer for this to be `string | boolean`. Then we validate that string is a supported compression.

I understand that probably `gzip` is the only option right now but it would be nice to have this be future proof.

Reply:
updated!